src/pool-bin/libraries/BinHelper.sol

```diff
@@ -148,31 +150,41 @@ library BinHelper {
     /// @dev Returns the composition fees when adding liquidity to the active bin with a different
     ///      composition factor than the bin's one, as it does an implicit swap
     /// @param binReserves The reserves of the bin
-    /// @param fee 100 = 0.01%, 1000 = 0.1%
+    /// @param protocolFee 100 = 0.01%, 1000 = 0.1%
+    /// @param lpFee 100 = 0.01%, 1000 = 0.1%
     /// @param amountsIn The amounts of tokens to add
     /// @param totalSupply The total supply of the liquidity book
     /// @param shares The share of the liquidity book that the user will receive
     /// @return fees The encoded fees that will be charged
     function getCompositionFees(
         bytes32 binReserves,
-        uint24 fee, // fee: 100 = 0.01%
+        uint24 protocolFee, // fee: 100 = 0.01%
+        uint24 lpFee,
         bytes32 amountsIn,
         uint256 totalSupply,
         uint256 shares
-    ) internal pure returns (bytes32 fees) {
-        if (shares == 0) return 0;
+    ) internal pure returns (bytes32 fees, bytes32 feeForProtocol) {
+        if (shares == 0) return (0, 0);
 
         (uint128 amountX, uint128 amountY) = amountsIn.decode();
         (uint128 receivedAmountX, uint128 receivedAmountY) =
             getAmountOutOfBin(binReserves.add(amountsIn), shares, totalSupply + shares).decode();
 
-        // if received more X than given X
+        // if received more X than given X, then swap some Y for X
         if (receivedAmountX > amountX) {
-            uint128 feeY = (amountY - receivedAmountY).getCompositionFee(fee);
+            protocolFee = protocolFee.getOneForZeroFee();
+
+            uint24 swapFee = uint24(protocolFee).calculateSwapFee(lpFee);
+            uint128 feeY = (amountY - receivedAmountY).getCompositionFee(swapFee);
             fees = feeY.encodeSecond();
+            feeForProtocol = (amountY - receivedAmountY).getCompositionFee(protocolFee).encodeSecond();
```

Comment:
Just a heads up:

The `getCompositionFees` here applies the following formula 
> `amountWithFees * feeBips * (1+feeBips) `

![image](https://github.com/pancakeswap/pancake-v4-core/assets/137024020/def29f38-75c6-457b-8762-a2e2fba9932a)

Not quite sure the reason, to make sure the calculation rule is the same i have used the same formula to calculate feeForProtocol here